doc/api/next_api_changes/behaviour.rst

```diff
@@ -93,3 +93,9 @@ Previously, `.SymLogNorm` had no *base* kwarg, and defaulted to ``base=np.e``
 whereas the documentation said it was ``base=10``.  In preparation to make
 the default 10, calling `.SymLogNorm` without the new *base* kwarg emits a
 deprecation warning.
+
+``add_subplot()`` validates its inputs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In particular, for ``add_subplot(rows, cols, index)``, all parameters must
+no be Integral. Previously strings and floats were accepted and converted to
+int. This will now raise an error.
```

Comment:
Are we ok to hard-raise on nonsensical inputs without a deprecation period?

I would say yes, because it's rather a bug that strings and floats were accepted.

Reply:
I think it better to warn here.  We had a test of taking floats, so something thought it was the correct behavior at one point.

Reply:
Changed to emitting a warning only.